File: easy/94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {Integer[]}
def inorder_traversal(root)
    inorder root, result = []
    result
end

def inorder(node, result)
    return if !node
    inorder node.left, result
    result << node.val
    inorder node.right, result
end

File: easy/94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
const inorderTraversal = function(root) {
    const res = [];
    
    (function dfs(node) {
        if (!node) return;
        
        dfs(node.left);
        res.push(node.val);
        dfs(node.right);
        
    }(root))
    
    return res;
};


//     4
//  2     5  
// 1 3   6 7 

// 1234567





